api/prisma/schema.prisma

```diff
@@ -116,6 +110,10 @@ model user {
   isCollegeAlgebraPyCertV8     Boolean? // Undefined
   isUpcomingPythonCertV8       Boolean? // Undefined
   keyboardShortcuts            Boolean? // Undefined
+  /// `lastUpdated` is used to track the last time the user's record was written to.
+  ///
+  /// This has the main benefit of allowing concurrent ops to check for race conditions.
+  lastUpdated                  Int                           @default(0)
```

Comment:
What's wrong with DateTime? It's a date, after all.

> The reason I did not want to do anything for create<>, is because, if we have any of the script logic on here, then whenever the script wanted to ask "have you updated since I last read your record", the answer will always be "yes".

Isn't this field supposed to do two things

1) tracking the last create/update to a given user
2) allowing updates to fail gracefully if the field changes between a read and an update

If I understand the $extends logic correctly, `lastUpdated` can play that role.

Reply:
I'm not 100% sure we're talking about the same thing, but this is what I just tested:

```ts
        const { lastUpdated } = await fastify.prisma.user.findUniqueOrThrow({
          where: { id: req.user!.id }
        });

        const newUser = await fastify.prisma.user.update({
          where: { id: req.user!.id, lastUpdated },
          data: { rand: 123 }
        });

        console.log(
          'update changed lastUpdated?',
          newUser.lastUpdated?.getTime() != lastUpdated?.getTime()
        );

        const updatedUser = await fastify.prisma.user.findUniqueOrThrow({
          where: { id: req.user!.id }
        });

        console.log(
          'findUnique changed lastUpdated?',
          updatedUser.lastUpdated?.getTime() != newUser.lastUpdated?.getTime()
        );

        await fastify.prisma.user.update({
          where: { id: req.user!.id, lastUpdated },
          data: { rand: 456 }
        });

```

The `findUniqueOrThrow` didn't change `lastUpdated` and the second update threw as expected.

Reply:
With a default of `Date.now()`, any `find<>` call will return a different time, until `update<>` is called. This is wrong.

Reply:
That's not what `default` is for. Its job is to provide the default value when you create a new record. Until you `update` all existing values will return `null`.

Reply:
There's a bug with Int, though. If you both set the field to `Int`, not `Int?`, and give a default value then you get that default value back when you `find`. `DateTime` and `DateTime?` behave correctly, though.

I'll create a small repro and report it tomorrow.